pkg/store/proxy.go

```diff
@@ -119,13 +127,15 @@ func newRespCh(ctx context.Context, buffer int) (*ctxRespSender, <-chan *storepb
 	return &ctxRespSender{ctx: ctx, ch: respCh}, respCh, func() { close(respCh) }
 }
 
+// send writes response to sender channel
+// or just returns if sender context was timed out
 func (s ctxRespSender) send(r *storepb.SeriesResponse) {
-	select {
-	case <-s.ctx.Done():
-		return
-	case s.ch <- r:
+	// If context closed or deadline exceeded - just skip sending and return
```

Comment:
I understand the concern here but what if we will exceed the deadline of the context before `s.ch <- r` succeeds?